### PR TITLE
Segment Registers are UReg_SR

### PR DESCRIPTION
**Pull request by Minty-Meeo**
This is a small one.  I noticed while working on a different PR that UReg_SR is almost entirely unused.

### CODE REVIEW

File: Source/Core/Core/PowerPC/PowerPC.cpp [hunk 1]
```diff
@@ -148,10 +148,12 @@ void DoState(PointerWrap& p)
 static void ResetRegisters()
 {
   std::fill(std::begin(ppcState.ps), std::end(ppcState.ps), PairedSingle{});
-  std::fill(std::begin(ppcState.sr), std::end(ppcState.sr), 0U);
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/PowerPC.cpp:151]: This should be able to be 

```cpp
std::fill(std::begin(ppcState.sr), std::end(ppcState.sr), UReg_SR{});
```

no?

Comment (Minty-Meeo) [reply Source/Core/Core/PowerPC/PowerPC.cpp:151]: I tried that, but the copy assignment operator for Bitfields is deleted, so a lot of errors get thrown.

Comment (Minty-Meeo) [reply Source/Core/Core/PowerPC/PowerPC.cpp:151]: What about std::generate?

Comment (Minty-Meeo) [reply Source/Core/Core/PowerPC/PowerPC.cpp:151]: This might be fixed if my suggestion for Common's BitField goes through (writing it up atm)

Comment (Minty-Meeo) [reply Source/Core/Core/PowerPC/PowerPC.cpp:151]: Once PR #10847 is approved, I can use std::fill here once again.

**Pull request closed without merging**